### PR TITLE
rolling-update: detect init system properly

### PR DESCRIPTION
**Pull request by leseb**
Previous use of the stat module didn't work as expected so using the
linux stat command now.

Closes: #1286

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

File: infrastructure-playbooks/rolling_update.yml [hunk 1]
```diff
@@ -120,13 +119,13 @@
...
         args: id={{ ansible_hostname }}
-      when: is_upstart.stat.exists == True
+      when: is_upstart.rc == 0
 
     - name: start ceph mons with sysvinit
       service:
         name: ceph
         state: started
-      when: is_sysvinit.stat.exists == True
+      when: is_sysvinit.0 == 0
```
Comment (andrewschoen) [inline infrastructure-playbooks/rolling_update.yml]: I think you want ``is_sysvinit.rc == 0`` here

Comment (leseb) [inline infrastructure-playbooks/rolling_update.yml]: done

**Comment by andrewschoen** [comment]
Actually, should we be using ``ansible_service_mgr`` here instead?  It looks like it was added in https://github.com/ansible/ansible/commit/99d775a0c12ff16c53aa2a19860d9293a2ef5b3c

File: infrastructure-playbooks/rolling_update.yml [hunk 2]
```diff
@@ -94,13 +100,13 @@
         name: ceph-mon
         state: stopped
         args: id={{ ansible_hostname }}
-      when: is_upstart.stat.exists == True
+      when: is_upstart.rc == 0
```
Comment (andrewschoen) [inline infrastructure-playbooks/rolling_update.yml]: Need to remove ``.rc == 0``

File: infrastructure-playbooks/rolling_update.yml [hunk 3]
```diff
@@ -94,13 +100,13 @@
...
 
     - name: stop ceph mons with sysvinit
       service:
         name: ceph
         state: stopped
-      when: is_sysvinit.stat.exists == True
+      when: is_sysvinit.rc == 0
```
Comment (andrewschoen) [inline infrastructure-playbooks/rolling_update.yml]: Need to remove ``.rc == 0``

**Pull request merged**